src/rbac/default_role_manager.rs

```diff
@@ -0,0 +1,353 @@
+use std::cell::RefCell;
+use std::rc::Rc;
+use std::collections::HashMap;
+use crate::rbac::RoleManager;
+
+type MatchingFunc = fn(&str, &str) -> bool;
+
+#[derive(Clone)]
+pub struct DefaultRoleManager {
+    pub all_roles: HashMap<String, Rc<RefCell<Role>>>,
+    pub max_hierarchy_level: usize,
+    pub has_pattern: bool,
+    pub matching_func: Option<MatchingFunc>,
+}
+
+impl DefaultRoleManager {
+    pub fn new(max_hierarchy_level: usize) -> Self {
+        return DefaultRoleManager {
+            all_roles: HashMap::new(),
+            max_hierarchy_level,
+            has_pattern: false,
+            matching_func: None,
+        };
+    }
+
+    fn create_role(&mut self, name: &str) -> Rc<RefCell<Role>> {
+        self.all_roles
+            .entry(name.to_owned())
+            .or_insert(Rc::new(RefCell::new(Role::new(name.to_owned()))))
```

Comment:
`or_insert_with` takes a closure which would allow the `Role` to only be constructed if needed.

Reply:
Thanks, you are right, it's better to use `or_insert_with` here, this has been fixed by @GopherJ at https://github.com/casbin/casbin-rs/blob/master/src/rbac/default_role_manager.rs#L29